crates/swc_common/src/comments.rs

```diff
@@ -381,7 +381,13 @@ impl Comments for SingleThreadedComments {
     fn move_leading(&self, from: BytePos, to: BytePos) {
         let cmt = self.leading.borrow_mut().remove(&from);
 
-        if let Some(cmt) = cmt {
+        if let Some(mut cmt) = cmt {
+            // TODO: if let chain
+            if from < to {
+                if let Some(to_cmt) = self.leading.borrow_mut().remove(&to) {
```

Comment:
I got wrong result with these changes in concurrent mode.

Reply:
Hmm... I have no idea why concurrent mode fails

Reply:
Not sure why is this logic required

Reply:
I am trying to keep the order of comments.

Reply:
See https://github.com/swc-project/swc/commit/877a3a5f87b3ba72d1be6107ef536443070ef397

Reply:
I revert it and get the same error 😅.

Reply:
Seems like current CI failure is simple update issue?

Reply:
No. I test it on my machine, the generated results changed randomly every time.
I suspect there is something wrong with `Comments` and concurrent mode.

Reply:
Ah... I'll investigate, thanks!